hack/generated/pkg/genruntime/conditions/conditions.go

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package conditions
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: Hah, name...
+type Conditioner interface {
+	GetConditions() []Condition
+	SetConditions(conditions []Condition)
+}
+
+// ConditionSeverity expresses the severity of a Condition.
+type ConditionSeverity string
+
+const (
+	// ConditionSeverityError specifies that a failure of a condition type
+	// should be viewed as an error. Errors are fatal to reconciliation and
+	// mean that the user must take some action to resolve
+	// the problem before reconciliation will be attempted again.
+	ConditionSeverityError ConditionSeverity = "Error"
+
+	// ConditionSeverityWarning specifies that a failure of a condition type
+	// should be viewed as a warning. Warnings are informational. The operator
+	// may be able to retry through the warning without any action from the user, but
+	// in some cases user action to resolve the warning will be required.
+	ConditionSeverityWarning ConditionSeverity = "Warning"
+
+	// ConditionSeverityInfo specifies that a failure of a condition type
+	// should be viewed as purely informational. Things are working.
+	// This is the happy path.
+	ConditionSeverityInfo ConditionSeverity = "Info"
+
+	// ConditionSeverityNone specifies that there is no condition severity.
+	// For conditions which have positive polarity (Status == True is their normal/healthy state), this will set when Status == True
+	// For conditions which have negative polarity (Status == False is their normal/healthy state), this will be set when Status == False.
+	// Conditions in Status == Unknown always have a severity of None as well.
+	// This is the default state for conditions.
+	ConditionSeverityNone ConditionSeverity = ""
+)
+
+type ConditionType string
+
+const (
+	// ConditionTypeReady is a condition indicating if the resource is ready or not.
+	// A ready resource is one that has been successfully provisioned to Azure according to the
+	// resource Spec.
```

Comment:
I think our users would appreciate that, but I'm also a bit hesitant assuming that just because a resource is successfully provisioned in Azure that it is "ready to use".

What this means (at a technical level, for ARM resources) is that the resource we have sent to Azure was successfully provisioned. The long running provisioning operation completed and the resource is in the `Provisioned` state. I am not sure that we can authoritatively say more than that without having a deep understanding of the specific resource in question?

Reply:
I realized that this status is actually broader than ARM, because it'll be on all resources regardless of if they are ARM or not. I tried to craft something generic that was helpful while also leaving a bit of room for uncertainty... I came up with this:
> 	// ConditionTypeReady is a condition indicating if the resource is ready or not.
	// A ready resource is one that has been successfully provisioned to Azure according to the
	// resource spec. It has reached the goal state. This usually means that the resource is ready
	// to use, but the exact meaning of Ready may vary slightly from resource to resource. Resources with
	// caveats to Ready's meaning will call that out in the resource specific documentation.

Let me know if that sounds good or if you have suggestions to change it.

Reply:
That works for me.